css-text-4/Overview.bs

```diff
@@ -10720,14 +10720,11 @@ Text Spacing Character Classes</h4>
 
 	<dl>
 		<dt><dfn>ideographs</dfn>
-		<dd>Includes all [=typographic character units=] [[CSS-TEXT-3]] whose base character is listed below:
-			<ul>
-				<li>All characters in the range of U+3041 to U+30FF,
-					except those that belong to Unicode Punctuation [P*] [=general category=].
-				<li>CJK Strokes (U+31C0 to U+31EF).
-				<li>Katakana Phonetic Extensions (U+31F0 to U+31FF).
-				<li>All characters that have the Han [=script property=].
-			</ul>
+		<dd>Includes all [=typographic character units=] [[CSS-TEXT-3]]
+		whose base character:
+		* belongs to Unicode Letters [L*], Mark [M*], Symbols [S*], or Numbers [N*] [=general category=], and
+		* has the Han, Hiragana, or Katakana [=script property=], and
+		* is not categorized as East Asian Halfwidth (H) by [[!UAX11]]
```

Comment:
This looks good. I think we also need to exclude EAW=H from "non-ideographic letters/numerics" too, right?

Reply:
seems reasonable, yes.

Reply:
I compared the proposed definition with "[spacing property](https://github.com/w3c/jlreq/blob/gh-pages/docs/spacing_property/spacing_property.md)" which is under development for jlreq-d. Differences are shown below:

Characters that are in css 'ideograph' but excluded from 'J' class in spacing_property
1. ㋐ CIRCLED KATAKANA (sc=Katakana, gc=So)
2. ㌁ SQUARE KATAKANA and HIRAGANA (sc=Katakana/Hiragana, gc=So)
3. Han radicals (sc=Hani, gc=So)
4. U+16FF0 VIETNAMESE ALTERNATE READING MARK CA and U+16FF1 VIETNAMESE ALTERNATE READING MARK NHAY (gc=Mc; sc=Hani; ea=W)
5. U+16FE3 OLD CHINESE ITERATION MARK (gc=Lm; sc=Hani; ea=W)

Characters that are excluded from css 'ideograph' but in 'J' class in spacing_property
6. 〆、〼 (gc=Lo; sc=Common; ea=W)
7. U+30FC KATAKANA-HIRAGANA PROLONGED SOUND MARK,  U-3031-3034 VERTICAL KANA REPEAT MARKs  (gc=Lm; sc=Common; ea=W)
8. Idegrahpic characters other than Han Ideograph (Ideographic=Yes, sc=Kits/Nshu/Tang)

Here are comments on each of these based on the discussions done when we developed spacing_property.

1. We believe circled characters should be removed from CSS 'Ideograph'. They were a part of cl-19 (Kanji) in JLReq, but it was agreed to treat all circled, double circled, negative circled and square letters as symbols that do not generate aki space with any other characters (class 'O' in the spacing_property). The reasoning is that they are enclosed symbols more than they are numbers, katanaka, kanji or alphabet (e.g, ①Ⓐ㉄㉮㊀).
2. We believe squared Katakana / Hiragana should be removed from CSS 'Ideograph' for the same reason as 1.
3. We believe radicals should stay in 'Ideograph', i.e as-is. We want to update spacing_property to match it.
4. We do not have enough knowledge to tell if it should be or should not be a part of 'Ideograph'. Since they are used in combination with base Kanji, they do not need to be included perhaps?
5. We believe this character should stay in 'Ideograph', i.e as-is. We want to update spacing_property to match it.
6. Probably 〆〼 should be added to 'Ideograph'. We do not have a strong opinion on them.
7. As agreed, the PROLONGED SOUND MARK should be added. The same goes to U-3031-3034 VERTICAL KANA REPEAT MARKs.
8. Probably all ideographic characters should be added to 'Ideograph' because they all originated from Han ideograph. At the same time we do not have a strong opinion on them.

 I might provide updates as I just asked JLReq TF if they have further comments.

Reply:
Regarding excluding East Asian Halfwidth (H), I believe we can just say "and is categorised as East Asian Wide (W)" because all characters in this category is "ea=W".  What do you think?

Reply:
If we are to update [class "J" of the spacing_property](https://github.com/w3c/jlreq/blob/gh-pages/docs/spacing_property/spacing_property.md#j-アルファベットとの間に空間を必要とする文字--仮名や漢字) with above recommendations incorporated, it would look like:
[[[[:sc=Hiragana:][:sc=Katakana:][:sc=Common:]][:ideographic:]]&[:gc=L:]&[:ea=W:]]
[[[:sc=Hani:]]&[[:gc=L:][:gc=Nl:][:gc=So:]]]

Reply:
@kidayasuo Your comments look great, but it also looks like the ideal list is more complicated than I would like to add to a CSS spec. How about doing this in 2 steps:
1. We go with some simple class for now, so that browsers can ship earlier.
2. Someone voluteers to add the property to Unicode, so that the future spec can switch to it.

For the purpose of 1, I'm fine with the current definition, or add/modify some more, but I wish to keep it simple.

Another possibility for 1: @nt1m @vitorroriz Is it possible for you to provide the list used by iOS/macOS? If it's simple enough, it might be a good candidate for the purpopse of 1 above.

@kidayasuo Are you or someone at JLTF willing to volunteer talking to Unicode?

> Regarding excluding East Asian Halfwidth (H), I believe we can just say "and is categorised as East Asian Wide (W)" because all characters in this category is "ea=W". What do you think?

Whether "ea=W" includes "ea=F" or not is a bit ambiguous. The [UAX#11](https://unicode.org/reports/tr11/) reads to me it does, but ICU doesn't. I think it's safer to be explicit.

/cc @Clqsin45

Reply:
> * We go with some simple class for now, so that browsers can ship earlier.

How simpler should it be?

For most practical uses I think the proposed changes + excluding ones below would be sufficient. 

> ㋐ CIRCLED KATAKANA (sc=Katakana, gc=So)
> ㌁ SQUARE KATAKANA and HIRAGANA (sc=Katakana/Hiragana, gc=So)

Reply:
cc @fantasai

Reply:
@kidayasuo and I will work on a proposal to Unicode. This will be 2 steps; one to reach a consensus to define a new property, then on the data. Hopefully the first step shouldn't be too hard.